### PR TITLE
Add normalizer for GISCorps national vaccine providers

### LINKED ISSUE #121: normalize GisCorps national data
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/main/vaccine_feed_ingest/schema/schema.py).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch us/giscorps_vaccine_providers
    poetry run vaccine-feed-ingest parse us/giscorps_vaccine_providers
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize us/giscorps_vaccine_providers
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by MoralCode**
Work in progress. Current progress documented in the linked issue

fixes #121

### CODE REVIEW

**Comment by MoralCode** [comment]
i should add that finding a way to link source URLs that use the automated arcGIS fetcher (like the line `fetched_from_uri` when setting the source) may be another thing that needs to be done as its not particularly easy to get to the link when all you have is the serviceItemId from the fetcher

**Comment by MoralCode** [comment]
once/if #267 is merged this should be ready to go apart from the operating hours field which still isn't done. is this important enough/necessary to have before merging?

**Comment by MoralCode** [comment]
This should now be good to go. I wouldn't say it's 100% complete (most notably the opening hours still need to be parsed from the arbitrary strings. might just want to throw those into the notes for now). but it should at least work well enough to get the data points into the list.

**Comment by eliblock** [comment]
@MoralCode - when I run this locally I get `ModuleNotFoundError: No module named 'schema'`. It looks like this might not be importing the schema in the same way as other normalizers?

I'm going to push a couple of commits to update to the new logger and make sure the schema is included, then can take another look.

Note for reviewers: `us/*` is disabled in the `Run all-stages for modified runners` CI check, so this PR needs to be checked locally.

**Review by eliblock — CHANGES_REQUESTED** [review]
I rebased this + updated to the schema announced in #278, and pushed results here.

When I run this locally, I see several validation errors (for example, the `source` + `id` didn't pass validation - I fixed that as a sample).

Could you take a look at ensuring the remaining data passes validation, and then we can land this code? It is totally okay to add something basic to start and then improve it over time by parsing more data.

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (eliblock) [inline vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:26]: Are we confident that "walk-in" should be the default value for a location that is not `appt_only` (vs. unknown)

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (eliblock) [inline vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:61]: Do we plan to capture this in the future? If not, let's remove the method!

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (eliblock) [inline vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:97]: 😩

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (eliblock) [inline vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:196]: This looks good for potential follow-up!

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (eliblock) [inline vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:212]: `latitude` and `longitude` are both required if a `LatLng` is set - do we need the `if`s in here?

I see the note below to ask about how it's convoluted 🙃.

Since this is scraped from an ArcGIS source, I'm also not too concerned about these being null, so going to ignore this before merging.

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (eliblock) [inline vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:229]: 🤔 this looks like the `ak/arcgis` url.

Where was this data fetched from?

**Review by eliblock — CHANGES_REQUESTED** [review]
@MoralCode - I'm supportive of landing this normalizer and then continuing to improve over time!

One thing that we need to fix before merging is the `fetched_from_uri`. Could you up date that to the proper link... and then we'll be good to go?

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (MoralCode) [reply vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:61]: i dont believe thsi dataset ever had the intention of including this data. will remove

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (MoralCode) [reply vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:97]: yeah, this isnt the best but there were a couple URLS in the dataset that were causing problems (as i mentioned in the discord)

one is literally a physical real-world address (that gets caught by the check for spaces in the url below this), one ended in a # and for some reason that broke the validation, and then there was the one with backslashes in it

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (MoralCode) [reply vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:196]: yep, this is more hacky than anything else. might be ripe to find an address-parsing library similar to my library for opening hours and use that instead

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (MoralCode) [reply vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:212]: yeah, essentially the line after it is supposed to basically make the whole thing null if the "geometry" key is not set, but looking at this again, a try/catch may be better here

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (MoralCode) [reply vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:229]: this data was fetched from the GIScorps dataset. i didnt change this URL because all thats listed in the fetch/parse stages is the serviceItemId and i was feeling lazy and didnt want to try and get the link back out from that. I'll go find the link and swap it out

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (MoralCode) [reply vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:26]: was not aware unknown was an option. will update

File: vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py
Comment (MoralCode) [reply vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:196]: https://github.com/datamade/usaddress looks good for this. should this wait for another PR to make this review easier?

Comment (eliblock) [reply vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py:194]: I'm going to merge this - let's add address improvements in a followup!

**Review by eliblock — APPROVED** [review]
Thanks - let's get this data moving (and can improve it later! Please open followups!)

**Pull request merged**
